### PR TITLE
hotfix / include ptrMatrices in matrix-axis slices

### PR DESCRIPTION
**Pull request by ruairimoran**
Make sure ptrMatrices is included when slicing on matrix axis

### CODE REVIEW

File: include/tensor.cuh [hunk 1]
```diff
@@ -659,10 +661,6 @@ DTensor<T>::DTensor(const DTensor<T> &other, size_t axis, size_t from, size_t to
     m_d_data = other.m_d_data + offset;
     m_doDestroyData = false;
     m_doDestroyPtrMatrices = false;
-    if (axis != 2) {
```
Comment (alphaville) [inline include/tensor.cuh:662]: I removed this because I put `m_d_ptrMatrices = nullptr;` at the beginning.

File: include/tensor.cuh [hunk 2]
```diff
@@ -640,11 +640,13 @@ template<typename T>
 DTensor<T>::DTensor(const DTensor<T> &other, size_t axis, size_t from, size_t to) {
     if (from > to) throw std::invalid_argument("from > to");
     size_t offset = 0, len = to - from + 1;
+    m_d_ptrMatrices = nullptr;
     if (axis == 2) {
         offset = other.m_numRows * other.m_numCols * from;
         m_numRows = other.m_numRows;
         m_numCols = other.m_numCols;
         m_numMats = len;
+        m_d_ptrMatrices = other.m_d_ptrMatrices + from;
```
Comment (alphaville) [inline include/tensor.cuh:648]: This line should be here (inside `if (axis == 2)`)

**Review by alphaville — APPROVED** [review]
@ruairimoran I made a couple of changes - we can merge if you're happy with them.

**Review by ruairimoran — COMMENTED** [review]
lgtm

**Pull request merged**